data/npc/lib/npc.lua

```diff
@@ -132,3 +132,25 @@ end
 function Player.getTotalMoney(self)
 	return self:getMoney() + self:getBankBalance()
 end
+
+function isValidMoney(money)
+	return isNumber(money) and money > 0 and money < 4294967296
+end
+
+function getMoneyCount(string)
+	local b, e = string:find("%d+")
+	local money = b and e and tonumber(string:sub(b, e)) or -1
+	if isValidMoney(money) then
+		return money
+	end
+	return -1
+end
+
+function getMoneyWeight(money)
+	local gold = money
+	local crystal = math.floor(gold / 10000)
+	gold = gold - crystal * 10000
+	local platinum = math.floor(gold / 100)
+	gold = gold - platinum * 100
+	return (ItemType(2160):getWeight() * crystal) + (ItemType(2152):getWeight() * platinum) + (ItemType(2148):getWeight() * gold)
```

Comment:
@luanluciano93 is changing IDs to constant here necessary too?

Reply:
in my view, yes

Reply:
Indeed, if the constants are available, they would be clearer to use.